app/controllers/application_controller.rb

```diff
@@ -20,6 +20,25 @@ class ApplicationController < ActionController::Base
 
   layout 'application'
 
+  # A common method for controllers to find a Dor object.  It accepts any valid
+  # PID (druid:aa111aa1111) or DRUID (aa111aa1111).  DruidTools::Druid parses
+  # input and returns a uniform PID, which is input to Dor.find to obtain a Dor
+  # object.
+  # @see https://github.com/sul-dlss/druid-tools
+  # @param druid [String]
+  # @return object [ActiveFedora::Base] any Dor object
+  # @raises ArgumentError
+  def find_druid(druid)
+    druid = DruidTools::Druid.new(druid).druid
+    Dor.find(druid)
+  rescue ArgumentError
+    flash[:error] = "Invalid DRUID: #{druid}"
+    render status: 400, text: flash[:error] and return
+  rescue ActiveFedora::ObjectNotFoundError # => e
+    flash[:error] = 'Object does not exist in Fedora.'
+    render status: 404, text: flash[:error] and return
```

Comment:
All understandable reasons for it to get to the state it's in right now.

But all I'm saying is, I don't think there's a need to do error handling at all in this method.  I'd propose just removing both of the `rescue` statements and otherwise leaving things as they are.

My justification is that I think there are two situations where it's right to trap an exception:
1) You can recover from it.  Sometimes people see that as using exceptions for flow control, and some people don't like that.  I don't particularly mind it if it's not being used as an ersatz `GOTO`, but that's not what's happening here anyway.
2) You can log something in greater detail or show a better error message, then you still propagate the error up the stack.  But I don't think the error messages here are enough of an improvement over the default Rails error page to justify having controller behavior in a method that shouldn't be restricted to being called from controllers.  `ApplicationController` already recognizes `ActiveFedora::ObjectNotFoundError` and displays a special error message in that case, so handling it above seems redundant.  And you could do the same for `ArgumentError`, as suggested in my prior comment (essentially using the same error message as in your 400 template).

With that approach, you'd still be presenting specific error messages for those conditions, and you'd do it without repeating the error handling code throughout the individual controllers, and you'd do it without including error handling that assumes web presentation in code that might be used in a wider variety of situations.  That seems like an all around improvement to me.

Additionally, the methods that call `find_druid` are somewhat inconsistent as to whether they check the result and return if `find_druid` returns `nil`.  If `find_druid` just exited on the spot with the exception it ran into, which isn't recovered from anyway, that check would become unnecessary and could be removed (because that code wouldn't be reached in the case of an unretrievable druid).

Maybe the `DruidTools` thing should be in dor-services.  I'd be fine with it staying here, I'd be fine with having a more robust `find` method there.  I do agree with your concern that using `DruidTools` in `Dor.find` might be too presumptuous.  But regardless of what happens with the DruidTools parsing, I'm very much in favor of pulling the error handling from this method, even if no other changes were made, since i don't that removal would result in a real functional deficiency.